### PR TITLE
implement sparsemax kernel operator

### PR DESCRIPTION
**Pull request by AndreasMadsen**
- [x] sparsemax kernel
- [x] sparsemax gradient
- [x] sparsemax loss kernel
- [x] sparsemax loss graident
- [x] Makefile cleanup

### CODE REVIEW

**Comment by AndreasMadsen** [comment]
@FrederikWR @MarcoDalFarra review please.

There are some possible optimizations, such as:
- sparse labels, `delta_k` vector vs `q` matrix (I need to read up on tensorflow polymorphism).
- early stopping in sparsemax calculation (see TODO).
- Transpose distribution `(grad * (-q + p).T).T = (-q + p) * grad.T` (see TODO).
- Use Eigen C++ framework for sparsemax loss calculations.

but none of these are strictly necessary.

**Comment by AndreasMadsen** [comment]
I was a little worried that because the loss function is used as `loss(logits, sparsemax(logits), labels)` that the gradient of `loss` with respect to `logits` would calculate the gradient of `sparsemax` with respect to `logits`. I have now confirmed that is not the case, all is good :)
